src/sqlfluff/core/templaters/jinja.py

```diff
@@ -326,6 +326,18 @@ def _slice_template(cls, in_str: str) -> Iterator[RawFileSlice]:
             "raw_end": "block",
             "raw_begin": "block",
         }
+
+        # Using Jinja whitespace stripping (e.g. `{%-` or `-%}`) breaks the
+        # position markers between unlexed and lexed file. So let's ignore any
+        # request to do that before lexing, by replacing '-' with '+'
+        #
+        # Note: '+' is the default, so shouldn't really be needed but we
+        # explicitly state that to preserve the space for the missing '-' character
+        # so it looks the same.
+        in_str = in_str.replace("{%-", "{%+")
+        in_str = in_str.replace("-%}", "+%}")
+        in_str = in_str.replace("{#-", "{#+")
+        in_str = in_str.replace("-#}", "+#}")
```

Comment:
Looks like dbt doesn't need this workaround and inherits this, so need to stick an `if` check in here:

```suggestion
        #
        # Note dbt inherits this class and does not need this workaround
        if cls.name == "jinja":
            in_str = in_str.replace("{%-", "{%+")
            in_str = in_str.replace("-%}", "+%}")
            in_str = in_str.replace("{#-", "{#+")
            in_str = in_str.replace("-#}", "+#}")
```